internal/infraestructure/github-connection/github-connection.go

```diff
@@ -2,54 +2,168 @@ package github_connection
+	}
+	pullRequestNumber, err := strconv.Atoi(pullRequestNumberStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid GITHUB_PR_NUMBER: %w", err)
+	}
 
-	GithubClient := github.NewClient(nil).WithAuthToken(params.Token)
+	eventName := os.Getenv("GITHUB_EVENT_NAME")
+	if eventName == "" {
+		return nil, fmt.Errorf("GITHUB_EVENT_NAME environment variable is not set")
+	}
+	GithubClient := github.NewClient(nil).WithAuthToken(token)
 
 	return &GithubConnection{
-		client:            GithubClient,
-		RepositoryName:    params.RepositoryName,
-		RepoOwner:         params.RepoOwner,
-		PullRequestNumber: params.PullRequestNumber,
+		Client:            GithubClient,
+		RepositoryName:    githubRepositoryName,
+		RepoOwner:         repoOwner,
+		PullRequestNumber: pullRequestNumber,
+		EventName:         eventName,
 	}, nil
 }
 
-type GithubConnection struct {
-	client            *github.Client
-	RepositoryName    string
-	RepoOwner         string
-	PullRequestNumber int
+func (receiver *GithubConnection) GetEventName() string {
+	return receiver.EventName
 }
 
-func (receiver *GithubConnection) GetRepositoryName() string {
+func (receiver *GithubConnection) GetRepository() string {
 	return receiver.RepositoryName
 }
 
-func (receiver *GithubConnection) GetRepository() string {
-	return receiver.RepositoryName
+func (receiver *GithubConnection) GetPullRequestChanges() ([]FileChange, error) {
+	var ctx = context.Background()
+	var opt = &github.ListOptions{PerPage: 100}
+	var allAdditions []FileChange
+
+	for {
+		files, resp, err := receiver.Client.PullRequests.ListFiles(
+			ctx, receiver.RepoOwner, receiver.RepositoryName, receiver.PullRequestNumber, opt)
+		if err != nil {
+			return nil, fmt.Errorf("error listing pull request files: %w", err)
+		}
+		for _, file := range files {
+			if file.Patch == nil {
+				continue
+			}
+			additions := parseAdditions(file)
+			if len(additions.LineNumbers) > 0 {
+				allAdditions = append(allAdditions, additions)
+			}
+		}
+
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
+	}
+
+	return allAdditions, nil
+}
+
+func (receiver *GithubConnection) CreateComment(comment string) error {
+	// ctx := context.Background()
+
+	allFiles, err := receiver.GetPullRequestChanges()
+	if err != nil {
+		return fmt.Errorf("error getting commits changes: %w", err)
+	}
+
+	for _, file := range allFiles {
+		fmt.Printf("File: %s, RawURL: %s\n", file.Filename, file.Content)
+		// commentData := &github.PullRequestComment{
+		// 	CommitID: github.String(os.Getenv("GITHUB_SHA")),
+		// 	Path:     github.String(*file.BlobURL),
+		// 	Body:     github.String(comment),
+		// }
+		// var prComment, _, err = receiver.Client.PullRequests.CreateComment(
+		// 	ctx,
+		// 	receiver.RepoOwner,
+		// 	receiver.RepositoryName,
+		// 	receiver.PullRequestNumber,
+		// 	commentData)
+		// if err != nil {
+		// 	return fmt.Errorf("error PullRequests: %w", err)
+		// }
+	}
+
+	return nil
 }
```

Comment:
Code Review 2024-08-18 01:18:35 - New lines 56 to 112:

func (receiver *GithubConnection) GetPullRequestChanges() ([]FileChange, error) {
	var ctx = context.Background()
	var opt = &github.ListOptions{PerPage: 100}
	var allAdditions []FileChange

	for {
		files, resp, err := receiver.Client.PullRequests.ListFiles(
			ctx, receiver.RepoOwner, receiver.RepositoryName, receiver.PullRequestNumber, opt)
		if err != nil {
			return nil, fmt.Errorf("error listing pull request files: %w", err)
		}
		for _, file := range files {
			if file.Patch == nil {
				continue
			}
			additions := parseAdditions(file)
			if len(additions.LineNumbers) > 0 {
				allAdditions = append(allAdditions, additions)
			}
		}

		if resp.NextPage == 0 {
			break
		}
		opt.Page = resp.NextPage
	}

	return allAdditions, nil
}

func (receiver *GithubConnection) CreateComment(comment string) error {
	// ctx := context.Background()

	allFiles, err := receiver.GetPullRequestChanges()
	if err != nil {
		return fmt.Errorf("error getting commits changes: %w", err)
	}

	for _, file := range allFiles {
		fmt.Printf("File: %s, RawURL: %s\n", file.Filename, file.Content)
		// commentData := &github.PullRequestComment{
		// 	CommitID: github.String(os.Getenv("GITHUB_SHA")),
		// 	Path:     github.String(*file.BlobURL),
		// 	Body:     github.String(comment),
		// }
		// var prComment, _, err = receiver.Client.PullRequests.CreateComment(
		// 	ctx,
		// 	receiver.RepoOwner,
		// 	receiver.RepositoryName,
		// 	receiver.PullRequestNumber,
		// 	commentData)
		// if err != nil {
		// 	return fmt.Errorf("error PullRequests: %w", err)
		// }
	}

	return nil